lib/ansible/modules/cloud/vmware/vmware_vm_shell.py

```diff
@@ -168,36 +163,34 @@ def main():
     if not HAS_PYVMOMI:
         module.fail_json(changed=False, msg='pyvmomi is required for this module')
 
-    try:
-        p = module.params
-        datacenter_name = p['datacenter']
-        cluster_name = p['cluster']
-        folder = p['folder']
-        content = connect_to_api(module)
-
-        datacenter = None
-        if datacenter_name:
-            datacenter = find_datacenter_by_name(content, datacenter_name)
-            if not datacenter:
-                module.fail_json(changed=False, msg="datacenter not found")
-
-        cluster = None
-        if cluster_name:
-            cluster = find_cluster_by_name(content, cluster_name, datacenter)
-            if not cluster:
-                module.fail_json(changed=False, msg="cluster not found")
-
-        if p['vm_id_type'] == 'inventory_path':
-            vm = find_vm_by_id(content, vm_id=p['vm_id'], vm_id_type="inventory_path", folder=folder)
-        else:
-            vm = find_vm_by_id(content, vm_id=p['vm_id'], vm_id_type=p['vm_id_type'], datacenter=datacenter, cluster=cluster)
-
-        if not vm:
-            module.fail_json(msg='VM not found')
-
-        msg = execute_command(content, vm, p['vm_username'], p['vm_password'],
-                              p['vm_shell'], p['vm_shell_args'], p['vm_shell_env'], p['vm_shell_cwd'])
+    p = module.params
```

Comment:
I am not very fond of this, as it makes code harder to read without context. It's more important that modules are written consistently, rather than optimize typing for the writer IMO. (The original was doing it wrong, I know)